cmd/bosun/database/state_data.go

```diff
@@ -150,6 +193,45 @@ func (d *dataAccess) GetAllIncidents(ak models.AlertKey) ([]*models.IncidentStat
 	return d.incidentMultiGet(conn, ids)
 }
 
+func (d *dataAccess) getAllIncidentIds() ([]int64, error) {
+	conn := d.Get()
+	defer conn.Close()
+
+	summaries, err := redis.Strings(conn.Do("LRANGE", "allIncidents", 0, -1))
+	if err != nil {
+		return nil, slog.Wrap(err)
+	}
+	ids := make([]int64, len(summaries))
+	for i, sum := range summaries {
+		var err error
+		ids[i], err = strconv.ParseInt(strings.Split(sum, ":")[0], 0, 64)
+		if err != nil {
+			return nil, slog.Wrap(err)
+		}
+	}
+	return ids, nil
+}
+
+func (d *dataAccess) getAllIncidentIdsByKeys() ([]int64, error) {
+	conn := d.Get()
+	defer conn.Close()
+
+	summaries, err := redis.Strings(conn.Do("KEYS", "incidentById:*"))
+	if err != nil {
+		return nil, slog.Wrap(err)
+	}
+	ids := make([]int64, len(summaries))
+	for i, sum := range summaries {
+		var err error
+		ids[i], err = strconv.ParseInt(strings.Split(sum, ":")[1], 0, 64)
+		if err != nil {
+			return nil, slog.Wrap(err)
+		}
+	}
+	return ids, nil
+
+}
+
```

Comment:
Really we should just remove allIncidents altogether. It is not updated when you remove incidents. It does not look like we ever read from it. And you have shown we can get the incident data the hard way (with KEYS) if we need to.